spacepy/irbempy/__init__.py

```diff
@@ -83,11 +83,2472 @@
+    current_dates = []
+    for vals in current_days:
+        try:
+            adddate = spt.doy2date(int(vals.split('_')[0]), int(vals.split('_')[1]), dtobj=True)
+        except ValueError:  # not a valid date format
+            continue
+        current_dates.append(adddate)
+
+    request_days = spt.tickrange(firstDate, lastDate, 1)
+    timewant = set(request_days.UTC)
+    timegot = set(current_dates)
+    stillwant = sorted(list(timewant.difference(timegot)))
+
+    print("Retrieving {2} requested files [{0} - {1}]".format(firstDate.date().isoformat(),
+                                                              lastDate.date().isoformat(),
+                                                              len(stillwant)))
+    for day in stillwant:
+        doy = spt.Ticktock(day).DOY[0]
+        name = '{0}_{1:03d}'.format(day.year, doy)
+        target = os.path.join(path, 'Coeffs', name)
+        source = baseURL+'/{0}/'.format(day.year)+'{0}.tgz'.format(name)
+        targetfile = os.path.join('{0}.tgz'.format(name))
+        if verbose:
+            print("\nFetching archive for {0}".format(day.date().isoformat()))
+            tmp, report = u.urlretrieve(source, targetfile, reporthook=tb.progressbar)
+        else:
+            tmp, report = u.urlretrieve(source, targetfile)
+        # if the file existed on the server then we won't get an HTML message back
+        if 'html' not in report['content-type']:
+            tar = tarfile.open(tmp, 'r:gz')
+            tar.extractall(target)
+            tar.close()
+        # now remove tgz file
+        os.unlink(tmp)
+
+
+# -----------------------------------------------
+def get_Bfield(ticks, loci, extMag='T01STORM', options=[1, 0, 0, 0, 0], omnivals=None):
+    """
+    Return magnetic field vector (in GEO) and magnitude
+
+    Calls get_bfield from IRBEMlib and uses the underlying model
+    implementations and coordinate transforms in IRBEMlib to obtain the
+    result.
+
+    Parameters
+    ----------
+        - ticks (Ticktock class) : containing time information
+        - loci (Coords class) : containing spatial information
+        - extMag (string) : optional; will choose the external magnetic
+                            field model possible values ['0', 'MEAD',
+                            'T87SHORT', 'T87LONG', 'T89', 'OPQUIET',
+                            'OPDYN', 'T96', 'OSTA', 'T01QUIET',
+                            'T01STORM', 'T05', 'ALEX', 'TS07']
+        - options (optional list or array of integers length=5) : explained in Lstar
+        - omni values as dictionary (optional) : if not provided, will
+                                                 use OMNI module to
+                                                 look up
+        - (see Lstar documentation for further explanation)
+
+    Returns
+    -------
+        - results (dictionary) : containing keys: Bvec, and Blocal
+                                 Bvec is specified in GEO coordinates
```

Comment:
This diff looks bigger than it is because we were moving files. The docstrings are basically as original, and fixing them is for you in #759 (which is why we rebased that against this PR).